### PR TITLE
Interop/Stress client that is resilient to server restarts

### PR DESCRIPTION
**Pull request by sreecha**
- Made changes to the interop client to not abort() in case of connection failures. This is configurable via a new command line flag `do_not_abort_on_transient_failures` which is always set to `false` in case of interop client and `true` in case of stress client 

- Added support to the interop tests in stress client 
```
              "   server_compressed_streaming" 
              "   slow_consumer"
              "   half_duplex"
              "   ping_pong"
              "   cancel_after_begin"
              "   cancel_after_first_response"
              "   timeout_on_sleeping_server"
              "   status_code_and_message"
              "   custom_metadata"
```

- @ctiller made a small change in `subchannel_call_holder.c` to ensure that if send initial metadata operation fails with a connection failure, then send message operation also fails. 

**Note:** I manually tested all the interop operations via `interop_client` (to make sure I am not introducing any regression) and also `stress_client`.
I noticed that `server_compressed_streaming` and `timeout_on_sleeping_server` tests fail with the stress client. The `server_compressed_streaming` is a known issue but the `timeout_on_sleeping_server` failure is a new one . I will be opening a separate issue for it.

### CODE REVIEW

File: test/cpp/interop/client.cc [hunk 1]
```diff
@@ -81,6 +81,14 @@ DEFINE_string(default_service_account, "",
 DEFINE_string(service_account_key_file, "",
               "Path to service account json key file.");
 DEFINE_string(oauth_scope, "", "Scope for OAuth tokens.");
+DEFINE_bool(do_not_abort_on_transient_failures, false,
```
Comment (yang-g) [inline test/cpp/interop/client.cc]: nit: the flag name is long and has a not in it. Maybe remove do_not_ and change the default value?

**Comment by yang-g** [comment]
LGTM

File: test/cpp/interop/client.cc (hunk 1 above)
Comment (sreecha) [inline test/cpp/interop/client.cc]: Actually I  too do not like double-negative names like this one :( and I started off by naming this `abort_on_transient_failures` (with default value to `true`). However, I felt that the name `abort_on_transient_failures` (irrespective of its default value) gave the impression that abort was not called for non-transient failures - which is not true.
So I felt `do_not_abort_on_transient_failures` was a better name.. It does correctly seem to convey that normally abort is called on all failures and that we just do not want to call it on transient failures.

**Comment by sreecha** [comment]
The test failures look like known flakes. Could this be merged (?)

**Comment by jtattermusch** [comment]
I think it's too red. Running tests again.

**Pull request merged**